components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/MonitoredItemMessageEncoder.cs

```diff
@@ -316,30 +322,44 @@ public class MonitoredItemMessageEncoder : IMessageEncoder {
+                                MessageContentMask = (message.Writer?.MessageSettings?
                                .DataSetMessageContentMask).ToMonitoredItemMessageMask(
                                    message.Writer?.DataSetFieldContentMask),
-                            ApplicationUri = message.ApplicationUri,
-                            EndpointUrl = message.EndpointUrl,
-                            ExtensionFields = message.Writer?.DataSet?.ExtensionFields,
-                            NodeId = notification.NodeId.ToExpandedNodeId(message.ServiceMessageContext?.NamespaceUris),
-                            Timestamp = message.TimeStamp ?? DateTime.UtcNow,
-                            Value = notification.Value,
-                            DisplayName = notification.DisplayName,
-                            SequenceNumber = notification.SequenceNumber.GetValueOrDefault(0)
-                        };
-                        // force published timestamp into to source timestamp for the legacy heartbeat compatibility
-                        if (notification.IsHeartbeat &&
-                            ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.Timestamp) == 0) &&
-                            ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.SourceTimestamp) != 0)) {
-                            result.Value.SourceTimestamp = result.Timestamp;
+                                ApplicationUri = message.ApplicationUri,
+                                EndpointUrl = message.EndpointUrl,
+                                ExtensionFields = message.Writer?.DataSet?.ExtensionFields,
+                                NodeId = notification.NodeId.ToExpandedNodeId(message.ServiceMessageContext?.NamespaceUris),
+                                Timestamp = message.TimeStamp ?? DateTime.UtcNow,
+                                Value = notification.Value,
+                                DisplayName = notification.DisplayName,
+                                SequenceNumber = notification.SequenceNumber.GetValueOrDefault(0)
+                            };
+                            // force published timestamp into to source timestamp for the legacy heartbeat compatibility
+                            if (notification.IsHeartbeat &&
+                                ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.Timestamp) == 0) &&
+                                ((result.MessageContentMask & (uint)MonitoredItemMessageContentMask.SourceTimestamp) != 0)) {
+                                result.Value.SourceTimestamp = result.Timestamp;
+                            }
+
+                        }
+                        catch (Exception e) {
+                            _logger.Error(e, "Error in notification {notificationCounter}/{notificationCount}: {exception}",
+                                notificationCounter, message.Notifications.Count(), e.Message);
```

Comment:
Ah - sorry, what I meant was, I thought putting the exception message as result into a "dummy" encoded message or so.   Since we already log the entire exception, the message is not needed in the debug log.   We can add a backlog item to do so though.

Reply:
Understood, I created a backlog item for this